diff -up ypserv-2.19/rpc.yppasswdd/update.c.crypt ypserv-2.19/rpc.yppasswdd/update.c
--- ypserv-2.19/rpc.yppasswdd/update.c.crypt	2012-04-27 13:44:56.171826593 +0200
+++ ypserv-2.19/rpc.yppasswdd/update.c	2012-04-27 13:46:49.501684728 +0200
@@ -218,12 +218,25 @@ putspent_adjunct (const struct spwd *p,
 static int
 password_ok (char *plain, char *crypted, char *root)
 {
+  char *crypted_new;
   if (crypted[0] == '\0')
     return 1;
-  if (strcmp (crypt (plain, crypted), crypted) == 0)
+  crypted_new = crypt (plain, crypted);
+  if (crypted_new == NULL)
+    {
+      log_msg ("crypt() call failed.");
+      return 0;
+    }
+  if (strcmp (crypted_new, crypted) == 0)
     return 1;
 #if CHECKROOT
-  if (strcmp (crypt (plain, root), root) == 0)
+  crypted_new = crypt (plain, root);
+  if (crypted_new == NULL)
+    {
+      log_msg ("crypt() call failed.");
+      return 0;
+    }
+  if (strcmp (crypted_new, root) == 0)
     return 1;
 #endif
 
